src/app/subject/components/source-graphs/source-graph/source-graph.component.ts

```diff
@@ -66,7 +68,13 @@ export class SourceGraphComponent implements OnDestroy {
   graphMargins = { top: 32, right: 16, bottom: 32, left: 80 }
 
   get keys() {
-    return this.sourceData.keys
+    if (this.isSingle) {
+      if (AppConfig.config.units[this.sourceData.unit]) {
+        return [AppConfig.config.units[this.sourceData.unit].label.EN]
+      }
+    } else {
+      return this.sourceData.keys
```

Comment:
if it is not single, we should get the keys with corresponding units. Otherwise, we show units for single line charts and legends without units for multi-type charts.